ykrt/src/mt.rs

```diff
@@ -715,6 +715,44 @@ mod tests {
         ));
     }
 
+    #[test]
+    fn mark_locations_for_which_tracing_doesnt_succeed_as_donttrace() {
+        // If a thread starts tracing a Location and tracing does not stop before the thread
+        // terminates, that Location should be marked (currently forever!) as DontTrace.
+
+        const THRESHOLD: HotThreshold = 5;
+        let mt = MT::new();
+        mt.set_hot_threshold(THRESHOLD);
+        let loc = Arc::new(Location::new());
+
+        for _ in 0..THRESHOLD {
+            assert_eq!(mt.transition_location(&loc), TransitionLocation::NoAction);
+        }
+
+        {
+            let mt = mt.clone();
+            let loc = Arc::clone(&loc);
+            thread::spawn(move || {
+                assert!(matches!(
+                    mt.transition_location(&loc),
+                    TransitionLocation::StartTracing(_)
+                ));
+            })
+            .join()
+            .unwrap();
+        }
+
+        assert_eq!(
+            hotlocation_discriminant(&loc),
+            Some(HotLocationDiscriminants::Tracing)
+        );
+        assert_eq!(mt.transition_location(&loc), TransitionLocation::NoAction);
+        assert_eq!(
```

Comment:
I'd be tempted to repeat that last assertion once more.

Reply:
That won't really test anything meaningful.

Reply:
It tests (to a limited degree) that further calls to `transition_location` don't exit the "do not trace state".

I'm assuming that once a location is forsaken, it's forsaken for good.

Reply:
It is -- but this is something I'd like to change soon. [An additional problem is that testing it once more doesn't really prove much -- in a sense you have to test it infinitely often, which will make CI unhappy :)]